### ISSUE TITLE
`sdk outdated` updates only the first outdated candidate

### ISSUE THREAD

**Issue by hho** [sdkman/sdkman-cli#522]
When there's more than one update available, SDKMAN seems to only update the first one:

```
hho@hho-pc3: ~
➜ sdk o

Upgrade:
gradle (3.2.1 < 3.3)
kotlin (1.0.5-2 < 1.0.6)

Upgrade candidate(s) and set latest version(s) as default? (Y/n): y

Downloading: gradle 3.3

In progress...

######################################################################## 100,0%

Installing: gradle 3.3
Done installing!


Setting gradle 3.3 as default.

hho@hho-pc3: ~
➜ sdk c

Using:

asciidoctorj: 1.5.4.1
gradle: 3.3
groovy: 2.4.7
kotlin: 1.0.5-2
maven: 3.3.9

hho@hho-pc3: ~
➜
```

When I ran `sdk o` again, it correctly upgraded Kotlin. This is on beta channel, version `master+186`, using Ubuntu 16.10 (`GNU bash, version 4.3.46(1)-release (x86_64-pc-linux-gnu)`).

In case it's relevant: I do have sdkman sourced from `.profile` and it seems Ubuntu _is_ using a login shell.

**Comment by pbassiner** [comment]
Hi @hho,
I cannot reproduce it under the same conditions:
* Ubuntu 16.10
* beta channel enabled
* sdk version `master+186`
* having `gradle 3.2.1` and `kotlin 1.0.5-2`
* and running `sdk o` upgrades both of them

Could you try it again? You can remove the candidates using `sdk rm gradle 3.3`, for instance.

**Comment by hho** [comment]
Tried again, same result: https://asciinema.org/a/98782

**Comment by marc0der** [comment]
I think this might be a bash/zsh incompatibility. I've seen this sort of thing before, in particular with array handling. Could this be the issue?

**Comment by pbassiner** [comment]
Turns out it is. I've been able to reproduce it in `bash` (works fine in `zsh`).

**Comment by marc0der** [comment]
Great news!

I remember having to do a similar workaround in the init script for working with candidates. If I remember right, it had to do with one of the shells handling arrays with a non-zero based index.

**Comment by pbassiner** [comment]
Fix proposal available in PR #524

**Comment by marc0der** [comment]
@pbassiner thanks for another great PR! Will release it soon.
